clients/web/src/components/console/log-event/message.tsx

```diff
@@ -25,7 +53,113 @@ export function Message(props: {
             </span>
           </A>
         </Show>
-        {props.processedEvent.message}
+        <span
+          classList={{
+            "group block max-h-[60px]": true,
+            "overflow-hidden": collapse(),
+          }}
+        >
+          <span
+            classList={{
+              [`block relative z-50
+                before:shadow-[0_0_1rem_rgba(255,255,255,0.2)]
+                before:border-gray-800 before:border-2
+                before:absolute before:rounded-lg
+                before:bg-gray-950/90 before:left-[-1rem]
+                before:top-[-1rem] before:right-[-1rem]
+                before:bottom-[-1rem] before:z-0 before:pointer-events-none`]:
+                !collapse(),
+              "whitespace-pre-wrap": pre(),
+            }}
+            ref={setRef}
+          >
+            <span
+              classList={{
+                "block max-h-[50vh] overflow-auto relative": !collapse(),
+              }}
+            >
+              <Show when={message().length > 40} fallback={message()}>
+                {makeBreakable(message())}
+              </Show>
+            </span>
+
+            <span class="flex flex-row absolute bg-gray-950/50 border-box right-[-0.5rem] top-[-1rem] z-51 p-1 opacity-45 group-hover:opacity-100">
```

Comment:
You don't need to use custom values for these props.

- `right-[-0.5rem]` is the same as `-right-2`
- `top-[-1rem]` is the same as `-top-4`

But same as the comment above, probably.